simple-mdns/src/lib.rs

```diff
@@ -53,15 +56,15 @@ fn create_socket(addr: &SocketAddr) -> io::Result<Socket> {
     Ok(socket)
 }
 
-fn join_multicast(addr: &SocketAddr, interface: &Ipv4Addr) -> io::Result<UdpSocket> {
+fn join_multicast(addr: &SocketAddr, interface: &Interface) -> io::Result<UdpSocket> {
     let ip_addr = addr.ip();
 
     let socket = create_socket(addr)?;
 
     // depending on the IP protocol we have slightly different work
     match ip_addr {
         IpAddr::V4(ref mdns_v4) => {
-            socket.join_multicast_v4(mdns_v4, interface)?;
+            socket.join_multicast_v4(mdns_v4, &interface.0)?;
         }
         IpAddr::V6(ref mdns_v6) => {
             socket.join_multicast_v6(mdns_v6, 0)?;
```

Comment:
https://docs.rs/socket2/latest/socket2/struct.Socket.html#method.join_multicast_v6
> interface is the index of the interface to join/leave (or 0 to indicate any interface).

I am not sure here what 'any' means. Perhaps we need to find out the interface index somehow from the Ipv6Addr.
We could take the interface index from the user, but for convenience I like it that way. We need the Ipv6Addr anyhow in sender_socket() to bind. There are options to get the interface index to the IpAddr but it would add dependencies. Because I am not sure this works like intended, I havn't touched the docs yet.
What do you think?